tensorbay/client/dataset.py

```diff
@@ -829,10 +829,33 @@ def _upload_segment(
                     "All the frames should have the same patterns(all have frame id or not)."
                 )
 
+        done_set = {
+            frame.frame_id.timestamp().timestamp: frame for frame in segment_client.list_frames()
+        }
+
         if have_frame_id:
-            segment_filter = ((frame, None) for frame in segment)
+            segment_filter = ((frame, None, True) for frame in segment)
+            if skip_uploaded_files:
+                segment_filter = filter(
+                    lambda frame_dict: pbar.update_for_skip(
+                        not bool(done_set.get(frame_dict[0].frame_id.timestamp().timestamp))
```

Comment:
why add `bool`?

what if `timestamp` is 0?

`set.__contain__`? `key in set`?